dipy/viz/actor.py

```diff
@@ -539,3 +540,186 @@ def axes(scale=(1, 1, 1), colorx=(1, 0, 0), colory=(0, 1, 0), colorz=(0, 0, 1),
+                                             colormap=colormap,
+                                             global_cm=global_cm)
+            self.SetMapper(self.mapper)
+
+    odf_actor = OdfSlicerActor()
+    I, J, K = odfs.shape[:3]
+    odf_actor.display_extent(0, I, 0, J,
+                             int(np.floor(K/2)), int(np.floor(K/2 + 1)))
+
+    return odf_actor
+
+
+def _odf_slicer_mapper(odfs, affine=None, mask=None, sphere=None, scale=2.2,
+                       norm=True, radial_scale=True, opacity=1.,
+                       colormap=None, global_cm=False):
+    """ Helper function for slicing spherical fields
+
+    Parameters
+    ----------
+    odfs : ndarray
+        4D array of spherical functions
+    affine : array
+        4x4 transformation array from native coordinates to world coordinates
+    mask : ndarray
+        3D mask
+    sphere : Sphere
+        a sphere
+    scale : float
+        Distance between spheres.
+    norm : bool
+        Normalize `sphere_values`.
+    radial_scale : bool
+        Scale sphere points according to odf values.
+    opacity : float
+        Takes values from 0 (fully transparent) to 1 (non-transparent)
+    colormap : None or str
+        If None then no color is used.
+    global_cm : bool
+        If True the colormap will be applied in all ODFs. If False
+        it will be applied individually at each voxel (default False).
+    """
+    if mask is None:
+        mask = np.ones(odfs.shape[:3])
+
+    ijk = np.ascontiguousarray(np.array(np.nonzero(mask)).T)
+
+    if affine is not None:
+        ijk = np.ascontiguousarray(apply_affine(affine, ijk))
+
+    faces = np.asarray(sphere.faces, dtype=int)
+    vertices = sphere.vertices
+
+    all_xyz = []
+    all_faces = []
+    all_ms = []
+    for (k, center) in enumerate(ijk):
+
+        m = odfs[tuple(center.astype(np.int))].copy()
+
+        if norm:
+            m /= abs(m).max()
+
+        if radial_scale:
+            xyz = vertices * m[:, None]
+        else:
+            xyz = vertices.copy()
+
+        all_xyz.append(scale * xyz + center)
+        all_faces.append(faces + k * xyz.shape[0])
+        all_ms.append(m)
+
+    all_xyz = np.ascontiguousarray(np.concatenate(all_xyz))
+    all_xyz_vtk = numpy_support.numpy_to_vtk(all_xyz, deep=True)
+
+    all_faces = np.concatenate(all_faces)
+    all_faces = np.hstack((3 * np.ones((len(all_faces), 1)),
+                           all_faces))
+    ncells = len(all_faces)
+
+    all_faces = np.ascontiguousarray(all_faces.ravel(), dtype='i8')
+    all_faces_vtk = numpy_support.numpy_to_vtkIdTypeArray(all_faces,
+                                                          deep=True)
+    if global_cm:
+        all_ms = np.ascontiguousarray(
+            np.concatenate(all_ms), dtype='f4')
+
+    points = vtk.vtkPoints()
+    points.SetData(all_xyz_vtk)
+
+    cells = vtk.vtkCellArray()
+    cells.SetCells(ncells, all_faces_vtk)
+
+    if colormap is not None:
```

Comment:
Just take the reconst_csd.py example or even better your SFM example and replace sphere_funcs with odf_slicer. Then to change the view just select the ROI you want with the display_extent method.